_posts/2016-07-20-Automation-testing-with-postman.md

```diff
@@ -0,0 +1,166 @@
+---
+layout: post
+authors: [greg_rinaldi]
+title: 'API Testing with Postman and Newman'
+image: /img/postman.png
+tags: [Postman,Integration Testing,Newman,Automation,Testing]
+category: Testing
+comments: true
+---
+### Prerequisites
+For the purpose of this tutorial it is mandatory to have [Postman](https://www.getpostman.com/){:target="_blank"} installed which is available as a Chrome extension, as well as a Mac application. And it is also mandatory to create an account at [Algorithmia](https://www.algorithmia.com/){:target="_blank"}.
+
+### Creating and selecting an environment
+Postman's environment functionality makes it very easy to switch between different environments. A set of variables can be configured per environment and when switching thes variables will be repaced accordingly. It is very traight forwward for example let’s create an environment called "production".  Click the "No environment" dropdown in the header and select "Manage environments".
+<br/>
+<br/>
+![Alt text](/img/postman-2016/manage_environment.png)
+<br/>
+<br/>
+Select the “Add” button on the modal that is presented to you.  Add `url https://api.algorithmia.com/v1/algo/` and `key simNz9pf7hfAQNifdApf4K1GFhs1`.  Don't forget to replace the secret by your own key.
+ <br/>
+ <br/>
+![Alt text](/img/postman-2016/manage_environment_values.jpg)
+<br/>
+<br/>
+Finally select the "Production" environment in the environment dropdown.
+<br/>
+<br/>
+![Alt text](/img/postman-2016/manage_environment_production.png)
+
+### Creating a POST request
+First let's enter `{% raw %}{{url}}/WayneS/Calculator/0.1.0{% endraw %}` in the request field and change the method from GET to POST.  We also need some headers so add `Content-Type application/json`, `Authorization {% raw %}Simple {{key}}{% endraw %}` aswell.  Here we are using the environment variable `{% raw %}{{url}}{% endraw %}` and `{% raw %}{{key}}{% endraw %}` so that when we switch environments the variables are replaced accordingly.  The `{% raw %}{{...}}{% endraw %}` format can only be used in the request URL/URL params/Header values/form-data/url-encoded values/Raw body content/Helper fields.
+<br/>
+<br/>
+![Alt text](/img/postman-2016/request_headers.png)
+<br/>
+<br/>
+Postman also has a few dynamic variables which you can use. For example `{% raw %}{{$guid}}{% endraw %}` a v4 style guid, `{% raw %}{{$timestamp}}{% endraw %}` the current timestamp, `{% raw %}{{$randomInt}}{% endraw %}` a random integer between 0 and 1000. More of those will be added in future releases. But for now let us just simply enter `"x=log(2)"` as the raw content of our request.
```

Comment:
"For example {{$guid}} a v4 style guid".
For example, {{$guid}} is the v4 style guid.
But for now, let us just simply enter "x=log(2)" as the raw content of our request.